src/cryptography/hazmat/primitives/twofactor/totp.py

```diff
@@ -36,7 +36,7 @@ def __init__(
         self._time_step = time_step
         self._hotp = HOTP(key, length, algorithm, backend, enforce_key_length)
 
-    def generate(self, time: int):
+    def generate(self, time: typing.Union[int, float]):
```

Comment:
We have a test that confirms floats are allowed. We document int only though. For compatibility I defined it this way.